### ISSUE TITLE
Jupyter extension activation fails 

### ISSUE THREAD

**Issue by mlangiu** [microsoft/vscode#174376]
OS: Mac OS Catalina (10.15.7)
Version: 1.75.1
Commit: 441438abd1ac652551dbe4d408dfcec8a499b8bf
Date: 2023-02-08T21:34:59.000Z (5 days ago)
Electron: 19.1.9
Chromium: 102.0.5005.194
Node.js: 16.14.2
V8: 10.2.154.23-electron.0
OS: Darwin x64 19.6.0
Sandboxed: No

When opening a Python file or setting the language of a new file to Python, an attempt to load the Jupyter extension is triggered and fails (Running manual Jupyter notebooks from a system terminal works without problems) with the following error message being displayed at the bottom right corner:

> Extension activation failed run the 'Developer: Toggle Developer Tools' command for more information
>
>
> Source: Jupyter (Extension)

<details><summary>The associated console dump can be found here</summary>

    log.ts:398  WARN Via 'product.json#extensionEnabledApiProposals' extension 'github.vscode-pull-request-github' wants API proposal 'commentsResolvedState' but that proposal DOES NOT EXIST. Likely, the proposal has been finalized (check 'vscode.d.ts') or was abandoned.
    notificationsAlerts.ts:42 Extension activation failed, run the 'Developer: Toggle Developer Tools' command for more information.
    c @ notificationsAlerts.ts:42
    (anonymous) @ notificationsAlerts.ts:28
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    addNotification @ notifications.ts:204
    notify @ notificationService.ts:175
    (anonymous) @ mainThreadMessageService.ts:77
    d @ mainThreadMessageService.ts:42
    $showMessage @ mainThreadMessageService.ts:36
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    log.ts:404   ERR [Extension Host] extension activation failed TypeError: e.endsWith is not a function
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785581
      at Array.filter (<anonymous>)
      at new ra (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785567)
      at t.resolveInstance (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:518676)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520614
      at t.resolve (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520877)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506724
      at e._get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506490)
      at e.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:505133)
      at Hg.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:378419)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876375
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876752
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876760
      at DE (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:878320)
      at Ui.db (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14818)
      at Ui.cb (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14519)
      at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:12519
      at async f.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6818)
      at async f.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6781)
      at async f.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6238)
    console.ts:137 [Extension Host] extension activation failed TypeError: e.endsWith is not a function
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785581
      at Array.filter (<anonymous>)
      at new ra (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785567)
      at t.resolveInstance (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:518676)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520614
      at t.resolve (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520877)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506724
      at e._get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506490)
      at e.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:505133)
      at Hg.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:378419)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876375
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876752
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876760
      at DE (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:878320)
      at Ui.db (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14818)
      at Ui.cb (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14519)
      at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:12519
      at async f.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6818)
      at async f.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6781)
      at async f.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6238)
    I @ console.ts:137
    $logExtensionHostMessage @ mainThreadConsole.ts:39
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    console.ts:137 [Extension Host] rejected promise not handled within 1 second: TypeError: e.endsWith is not a function
    I @ console.ts:137
    $logExtensionHostMessage @ mainThreadConsole.ts:39
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    console.ts:137 [Extension Host] stack trace: TypeError: e.endsWith is not a function
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785581
      at Array.filter (<anonymous>)
      at new ra (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785567)
      at t.resolveInstance (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:518676)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520614
      at t.resolve (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520877)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506724
      at e._get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506490)
      at e.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:505133)
      at Hg.get (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:378419)
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876375
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876752
      at /Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876760
      at DE (/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:878320)
      at Ui.db (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14818)
      at Ui.cb (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14519)
      at /Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:12519
      at async f.n (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6818)
      at async f.m (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6781)
      at async f.l (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6238)
    I @ console.ts:137
    $logExtensionHostMessage @ mainThreadConsole.ts:39
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    mainThreadExtensionService.ts:79 [[object Object]]e.endsWith is not a function
    $onExtensionRuntimeError @ mainThreadExtensionService.ts:79
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    mainThreadExtensionService.ts:80 TypeError: e.endsWith is not a function
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785581
      at Array.filter (<anonymous>)
      at new ra (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:1785567)
      at t.resolveInstance (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:518676)
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520614
      at t.resolve (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:520877)
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506724
      at e._get (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:506490)
      at e.get (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:2:505133)
      at Hg.get (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:378419)
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876375
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876752
      at vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:876760
      at DE (vscode-file://vscode-app/Users/marcolangiu/.vscode/extensions/ms-toolsai.jupyter-2023.1.2010391206/out/extension.node.js:17:878320)
      at Ui.db (vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14818)
      at Ui.cb (vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:14519)
      at vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:108:12519
      at async f.n (vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6818)
      at async f.m (vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6781)
      at async f.l (vscode-file://vscode-app/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/api/node/extensionHostProcess.js:94:6238)
    $onExtensionRuntimeError @ mainThreadExtensionService.ts:80
    N @ rpcProtocol.ts:455
    M @ rpcProtocol.ts:440
    H @ rpcProtocol.ts:370
    G @ rpcProtocol.ts:296
    (anonymous) @ rpcProtocol.ts:161
    invoke @ event.ts:701
    deliver @ event.ts:908
    fire @ event.ts:869
    fire @ ipc.net.ts:671
    Z.onmessage @ localProcessExtensionHost.ts:615
    log.ts:392  INFO [perf] Render performance baseline is 35ms
</details>

Additionally, when running any Jupyter-related commands such as `Jupyter: Create Interactive Window` I get error messages of the following type:

> Command 'Jupyter: Create Interactive Window' resulted in an error (command 'jupyter.createnewinteractive' not found)

I already tried deleting the `~/.vscode` folder, and only installing the Python and Jupyter extensions, both in the current release as well as the pre-release and older versions (latest from 2022), none of wich resolved the issue.

**Comment by lovettchris** [comment]
Same for me on Windows, I also cannot open Developer Tools any more... same version of VS code, using Python 3.8.  Jupyter notebook succeeds in finding my python environment (using conda) but I get an infinite progress bar at the top of the window and my Jupyter Output window contains:

```
Visual Studio Code (1.75.1, undefined, desktop)
Jupyter Extension Version: 2023.1.2010391206.
Python Extension Version: 2023.2.0.
Workspace folder d:\temp\face_segmentation
User belongs to experiment group 'jupyterTestcf'
User belongs to experiment group 'jupyterEnhancedDataViewer'
info 13:28:14.205: LSP Notebooks experiment is enabled
info 13:28:14.531: Find preferred kernel for d:\temp\face_segmentation\test.ipynb with metadata {"orig_nbformat":4,"language_info":{"name":"python"}} & preferred interpreter c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe
info 13:28:14.531: Find preferred kernel for d:\temp\face_segmentation\test.ipynb with metadata {"orig_nbformat":4,"language_info":{"name":"python"}} & preferred interpreter c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe
info 13:28:14.545: Find preferred kernel for d:\temp\face_segmentation\test.ipynb with metadata {"orig_nbformat":4,"language_info":{"name":"python"}} & preferred interpreter c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe
info 13:28:14.556: Find preferred kernel for d:\temp\face_segmentation\test.ipynb with metadata {"orig_nbformat":4,"language_info":{"name":"python"}} & preferred interpreter c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe
info 13:28:16.451: Preferred kernel .jvsc74a57bd0799abcba35f70097d02fca042963180a03ec3451fe1b7671ac5d22383cd0232c.c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe.c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe.-m#ipykernel_launcher is exact match or top match for non python kernels, (false, [object Object], false, false)
info 13:28:16.452: PreferredConnection: .jvsc74a57bd0799abcba35f70097d02fca042963180a03ec3451fe1b7671ac5d22383cd0232c.c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe.c:\Users\clovett.REDMOND\Anaconda3\envs\archai\python.exe.-m#ipykernel_launcher found for NotebookDocument: d:\temp\face_segmentation\test.ipynb
info 13:28:16.673: Process Execution: > ~\Anaconda3\envs\archai\python.exe -m pip list
> ~\Anaconda3\envs\archai\python.exe -m pip list
```
So maybe it is stuck on the pip list, when I run this myself in my own terminal it responds just fine.  Probably because of this infinite progress lock, I cannot bring up Developer Tools.  I'm using these versions of jupyter things:

```
jupyter                  1.0.0
jupyter_client           8.0.2
jupyter-console          6.5.1
jupyter_core             5.2.0
jupyter-events           0.6.3
jupyter_server           2.2.1
jupyter_server_terminals 0.4.4
jupyterlab-pygments      0.2.2
jupyterlab-widgets       3.0.5
ipykernel                6.21.2
ipython                  8.10.0
ipython-genutils         0.2.0
ipywidgets               8.0.4
scipy                    1.10.0
```
I've tried all my other older conda environments, none of them work today.  I use jupyter notebooks every day in vscode, so something is very broken....  strangely I get the same infinite progress bar even with `code . --disable-extensions` as shown 
below:

![image](https://user-images.githubusercontent.com/18707114/218868177-8fc2f279-f80f-4dc2-93a8-3b1dbc0b98a6.png)

**Comment by lovettchris** [comment]
Ah, interesting, a complete uninstall/reinstall of VS code resolved the issue for me.  Seems uninstall/reinstall does not remove all the extensions, so it kept all my extensions and the empty jupyter notebook opens just fine and infinite progress is gone:

![image](https://user-images.githubusercontent.com/18707114/218869157-7a1d707f-0b76-4d29-8d7f-bf2023b4a874.png)

This is what was supposed to happen after the pip list line that got stuck:

```
info 13:37:57.464: Process Execution: > ~\Anaconda3\envs\coyote\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"356cd457-929f-418c-b95a-144ff172be7f" --shell=9002 --transport="tcp" --iopub=9004 --f=c:\Users\clovett.REDMOND\AppData\Roaming\jupyter\runtime\kernel-v2-42076iOgwHZ93zVUc.json
> ~\Anaconda3\envs\coyote\python.exe -m ipykernel_launcher --ip=127.0.0.1 --stdin=9003 --control=9001 --hb=9000 --Session.signature_scheme="hmac-sha256" --Session.key=b"356cd457-929f-418c-b95a-144ff172be7f" --shell=9002 --transport="tcp" --iopub=9004 --f=c:\Users\clovett.REDMOND\AppData\Roaming\jupyter\runtime\kernel-v2-42076iOgwHZ93zVUc.json
info 13:37:57.464: Process Execution: cwd: d:\temp\face_segmentation
cwd: d:\temp\face_segmentation
info 13:37:57.681: ipykernel version & path 6.15.2, ~\Anaconda3\envs\coyote\lib\site-packages\ipykernel\__init__.py for c:\Users\clovett.REDMOND\Anaconda3\envs\coyote\python.exe
info 13:37:58.730: Started Kernel coyote (Python 3.10.6) (pid: 47764)
info 13:37:58.761: Process Execution: > ~\Anaconda3\envs\coyote\python.exe ~\.vscode\extensions\ms-toolsai.jupyter-2023.1.2010391206\pythonFiles\printJupyterDataDir.py
> ~\Anaconda3\envs\coyote\python.exe ~\.vscode\extensions\ms-toolsai.jupyter-2023.1.2010391206\pythonFiles\printJupyterDataDir.py
```

**Comment by mlangiu** [comment]
The uninstallation and installation pointed me to a solution!

On Mac systems VS code stores data in `$HOME/Library/Application Support/Code`, so a full reinstallation would need to delete this folder.
This indeed resolved the issue for me, but to avoid redoing all of my setup I narrowed down the exact cause cause by selectively deleting parts of that folder. In my case the culprit was the `settings.json` file in the `User` subdirectory.
Particularly, I had the lines 
```json
"jupyter.experiments.optInto": [
    false
    ],
```
in there.
I don't remember whether I selected an appropriate setting at some point.
Possibly this might be an outdated setting from an older version of the jupyter extension.
In any case, deleting these lines resolves this issue.
